pkg/tsdb/cloudwatch/log_actions.go

```diff
@@ -0,0 +1,318 @@
+	})
+
+	for _, event := range logEvents.Events {
+		messages = append(messages, event.Message)
+		timestamps = append(timestamps, event.Timestamp)
+	}
+
+	timestampField := data.NewField("ts", nil, timestamps)
+	timestampField.SetConfig(&data.FieldConfig{Title: "Time"})
+
+	messageField := data.NewField("line", nil, messages)
+
+	return data.NewFrame("logEvents", timestampField, messageField), nil
+}
+
+func (e *CloudWatchExecutor) handleDescribeLogGroups(ctx context.Context, logsClient cloudwatchlogsiface.CloudWatchLogsAPI, parameters *simplejson.Json) (*data.Frame, error) {
+	logGroupNamePrefix := parameters.Get("logGroupNamePrefix").MustString("")
+	var response *cloudwatchlogs.DescribeLogGroupsOutput = nil
+	var err error
+
+	if len(logGroupNamePrefix) < 1 {
+		response, err = logsClient.DescribeLogGroupsWithContext(ctx, &cloudwatchlogs.DescribeLogGroupsInput{
+			Limit: aws.Int64(parameters.Get("limit").MustInt64(50)),
+		})
+	} else {
+		response, err = logsClient.DescribeLogGroupsWithContext(ctx, &cloudwatchlogs.DescribeLogGroupsInput{
+			Limit:              aws.Int64(parameters.Get("limit").MustInt64(50)),
+			LogGroupNamePrefix: aws.String(logGroupNamePrefix),
+		})
+	}
+
+	if err != nil || response == nil {
+		return nil, err.(awserr.Error)
+	}
+
+	logGroupNames := make([]*string, 0)
+	for _, logGroup := range response.LogGroups {
+		logGroupNames = append(logGroupNames, logGroup.LogGroupName)
+	}
+
+	groupNamesField := data.NewField("logGroupName", nil, logGroupNames)
+	frame := data.NewFrame("logGroups", groupNamesField)
+
+	return frame, nil
+}
+
+func (e *CloudWatchExecutor) executeStartQuery(ctx context.Context, logsClient cloudwatchlogsiface.CloudWatchLogsAPI, parameters *simplejson.Json, timeRange *tsdb.TimeRange) (*cloudwatchlogs.StartQueryOutput, error) {
+	startTime, err := timeRange.ParseFrom()
+	if err != nil {
+		return nil, err
+	}
+
+	endTime, err := timeRange.ParseTo()
+	if err != nil {
+		return nil, err
+	}
+
+	if !startTime.Before(endTime) {
+		return nil, fmt.Errorf("Invalid time range: Start time must be before end time")
+	}
+
+	startQueryInput := &cloudwatchlogs.StartQueryInput{
+		StartTime:     aws.Int64(startTime.Unix()),
+		EndTime:       aws.Int64(endTime.Unix()),
+		Limit:         aws.Int64(parameters.Get("limit").MustInt64(1000)),
+		LogGroupNames: aws.StringSlice(parameters.Get("logGroupNames").MustStringArray()),
+		QueryString:   aws.String("fields @timestamp,@log,@logStream|" + parameters.Get("queryString").MustString("")),
+	}
+	return logsClient.StartQueryWithContext(ctx, startQueryInput)
+}
+
+func (e *CloudWatchExecutor) handleStartQuery(ctx context.Context, logsClient cloudwatchlogsiface.CloudWatchLogsAPI, parameters *simplejson.Json, timeRange *tsdb.TimeRange, refID string) (*data.Frame, error) {
+	startQueryResponse, err := e.executeStartQuery(ctx, logsClient, parameters, timeRange)
+	if err != nil {
+		return nil, fmt.Errorf(err.(awserr.Error).Message())
```

Comment:
Why not return the original error? What's the benefit of converting?

Reply:
Returning the original error would mean getting the following
`MalformedQueryException: mismatched input '@' expecting <EOF> (Service: AWSLogs; Status Code: 400; Error Code: MalformedQueryException; Request ID: 231d865b-79d4-4489-a6aa-23a64f8c3f1e) { Message_: "mismatched input '@' expecting <EOF> (Service: AWSLogs; Status Code: 400; Error Code: MalformedQueryException; Request ID: 231d865b-79d4-4489-a6aa-23a64f8c3f1e)" }`

rather than
`mismatched input '@' expecting <EOF> (Service: AWSLogs; Status Code: 400; Error Code: MalformedQueryException; Request ID: 4ee346e9-776a-49df-aac0-fbb7be125d23)`

Personally I prefer the second message since there is no redundant information as there is with the first error.

Reply:
I think there's a risk of losing valuable information from the original error by converting it like this. You don't know if the error format might change in the future, and you lose the error type (which might be useful to a caller). How about returning a wrapped error instead? 
```suggestion
		return nil, errutil.Wrap(err.(awserr.Error).Message(), err)
```